### ISSUE TITLE
Intermittent Errno::ECONNREFUSED from ddtrace when running rspec

### ISSUE THREAD

**Issue by arich** [DataDog/dd-trace-rb#3137]
Hello! Starting a separate thread split from here: https://github.com/DataDog/dd-trace-rb/issues/2955#issuecomment-1720147325

**Current behaviour**
Occasionally during tests in CI, we hit an unexpected error caused by ddtrace attempting to send a trace. DD_TRACE_TEST_MODE_ENABLED is not set, and should default to false.

Output: 
```
.............E, [2023-09-13T13:55:15.816276 #25] ERROR -- ddtrace: [ddtrace] (/usr/local/bundle/gems/ddtrace-1.14.0/lib/ddtrace/transport/http/client.rb:37:in `rescue in send_request') Internal error during Datadog::Transport::HTTP::Client request. Cause: Errno::ECONNREFUSED Failed to open TCP connection to 127.0.0.1:8126 (Connection refused - connect(2) for "127.0.0.1" port 8126) Location: /usr/local/lib/ruby/3.2.0/net/http.rb:1271:in `initialize'
```

We've hit this twice recently and it's contributing to our test suite failing unexpectedly without cause. 

We are not running datadog agent locally or in CI, but I would not expect it is needed, as I'm not planning to submit traces to datadog for local development or CI test runs.

**Expected behaviour**
ddtrace should not be running when `RAILS_ENV=test`. No traces should be sent from CI/test runs by default. 

**Steps to reproduce**
I'm not sure how to reproduce this! I can tell you about our environment below. We're running tests in a docker container in github actions, running tests with knapsack pro to split tests between multiple nodes. 

I was also able to reproduce this locally running rspec outside a docker container, however it did not fail the test in this case, it just output the same string to STDOUT. It's much less disruptive when the spec doesn't fail. 

Local output:
```
rspec
Running via Spring preloader in process 77916
Run options: exclude {:skip_on_ci=>false}

Randomized with seed 1992
............Capybara starting Puma...
* Version 6.3.1 , codename: Mugi No Toki Itaru
* Min threads: 0, max threads: 4
* Listening on http://127.0.0.1:2999
* Listening on http://[::1]:2999
* Listening on http://127.0.2.2:2999
* Listening on http://127.0.2.3:2999
.......E, [2023-09-15T10:24:31.191992 #77766] ERROR -- ddtrace: [ddtrace] (/Users/myname/.asdf/installs/ruby/3.2.2/lib/ruby/gems/3.2.0/gems/ddtrace-1.14.0/lib/ddtrace/transport/http/client.rb:37:in `rescue in send_request') Internal error during Datadog::Transport::HTTP::Client request. Cause: Errno::ECONNREFUSED Failed to open TCP connection to 127.0.0.1:8126 (Connection refused - connect(2) for "127.0.0.1" port 8126) Location: /Users/myname/.asdf/installs/ruby/3.2.2/lib/ruby/3.2.0/net/http.rb:1271:in `initialize'
..........
```

**Environment**

* **ddtrace version:** 1.14.0
* **Configuration block (`Datadog.configure ...`):**
```ruby
unless Rails.env.development? || Rails.env.test? # maybe this is related? Originally running datadog configure in test mode had issues.
  Datadog.configure do |c|
    c.tracing.instrument :faraday, split_by_domain: true
    c.tracing.instrument :http, split_by_domain: true
    c.tracing.instrument :rails, log_injection: true, service_name: "rails"
    c.tracing.instrument :redis
    c.tracing.instrument :rest_client
    c.tracing.instrument :sidekiq, log_injection: true, service_name: "sidekiq"
    c.tracing.instrument :pg, comment_propagation: "full", service_name: "postgres"

    c.version = APP_VERSION
  end

  Datadog::Tracing.before_flush(
    # Do not trace health checks to datadog
    Datadog::Tracing::Pipeline::SpanFilter.new { |span| span.name == "rack.request" && span.get_tag("http.url") == "/health" }
  )
end
```
* **Ruby version:** 3.2.2
* **Operating system:** Docker: ruby:3.2.2-alpine3.16 as base image, Mac OS 13.5.2 (22G91) locally
* **Relevant library versions:**
* Rails 7.0.6
* rspec 3.12.0

**Comment by ivoanjo** [comment]
In https://github.com/DataDog/dd-trace-rb/pull/3062 we fixed the autodetection code to not run ddtrace on `RAILS_ENV=test`. Hopefully this will fix the issue :)

**Comment by mcg** [comment]
@ivoanjo Any idea when we will see a new Gem cut with this fix?

**Comment by marcotc** [comment]
👋 @mcg, we are looking to release a new version next week.

**Comment by marcotc** [comment]
Thank you for your patience, the fix in #3062 has been released in [v1.15.0](https://github.com/DataDog/dd-trace-rb/releases/tag/v1.15.0).

Please let us know if the issue persists.
